test/functional/test-extension-analytics.js

```diff
@@ -54,36 +54,41 @@ describes.realWin('extension-analytics', {
-      win.document.body.appendChild(ele);
-      const baseEle = new BaseElement(ele);
-      registerServiceBuilderForDoc(
-          ampdoc, 'amp-analytics-instrumentation', MockInstrumentation);
-      // Force instantiation
-      getServiceForDoc(ampdoc, 'amp-analytics-instrumentation');
-      const config = {
-        'requests': {
-          'pageview': 'https://example.com/analytics',
-        },
-        'triggers': {
-          'trackPageview': {
-            'on': 'visible',
-            'request': 'pageview',
+    [true, false].forEach(disableImmediate => {
+      it('should create analytics element if analytics is installed, ' +
+          `disableImmediate ${disableImmediate}`, () => {
+        const config = {
+          'requests': {
+            'pageview': 'https://example.com/analytics',
           },
-        },
-      };
-      expect(baseEle.element.querySelector('amp-analytics')).to.be.null;
-      expect(insertAnalyticsElement(baseEle.element, config, true)).to.be.ok;
-      return timer.promise(50).then(() => {
-        const analyticsEle = baseEle.element.querySelector('amp-analytics');
-        expect(analyticsEle).to.not.be.null;
-        expect(analyticsEle.getAttribute('sandbox')).to.equal('true');
-        expect(analyticsEle.getAttribute('trigger')).to.equal('immediate');
-        const script = (analyticsEle).querySelector('script');
-        expect(script.textContent).to.jsonEqual(JSON.stringify(config));
-        expect(analyticsEle.CONFIG).to.jsonEqual(config);
-        expect(analyticsEle.getAttribute('sandbox')).to.equal('true');
+          'triggers': {
+            'trackPageview': {
+              'on': 'visible',
+              'request': 'pageview',
+            },
+          },
+        };
+        const ele = win.document.createElement('div');
+        win.document.body.appendChild(ele);
+        const baseEle = new BaseElement(ele);
+        registerServiceBuilderForDoc(
+            ampdoc, 'amp-analytics-instrumentation', MockInstrumentation);
+        // Force instantiation
+        getServiceForDoc(ampdoc, 'amp-analytics-instrumentation');
+        expect(baseEle.element.querySelector('amp-analytics')).to.be.null;
+        expect(insertAnalyticsElement(
+            baseEle.element, config, true, disableImmediate)).to.be.ok;
+        return timer.promise(50).then(() => {
+          const analyticsEle = baseEle.element.querySelector('amp-analytics');
+          expect(analyticsEle).to.not.be.null;
+          expect(analyticsEle.getAttribute('sandbox')).to.equal('true');
+          expect(analyticsEle.getAttribute('trigger')).to.equal(
+              disableImmediate ? '' : 'immediate');
+          const script = (analyticsEle).querySelector('script');
+          expect(script.textContent).to.jsonEqual(JSON.stringify(config));
+          expect(analyticsEle.CONFIG).to.jsonEqual(config);
+          expect(analyticsEle.getAttribute('sandbox')).to.equal('true');
+        });
```

Comment:
I see how this is checking whether the `amp-analytics` element has `trigger=immediate`, but this test wouldn't fail if that attribute weren't being properly respected. The `does start initialization when requested` test in `test-amp-analytics.js` seems to be checking that `trigger=immediate` -> immediate load; what would you think of extending that test to also verify that `trigger=""` does not lead to an immediate load?

Reply:
That would go beyond unit to functional testing IMO.  Right now we're relying on that fact that each item has proper test coverage and so long as a given test ensures it properly calls the API of another section of code, it is assumed that low-level code has coverage.

Reply:
Sorry, I wasn't proposing adding a functional test.  I was trying to point out that the code you're calling doesn't unit test the "no trigger immediate -> no immediate load" case, and since you're going to be the first consumer of it (I think?) it could be worth adding that testing.